### ISSUE TITLE
Db funcrions

### ISSUE THREAD

**Issue by stRavens** [planck/planck#15]
All functions for work with db can be divided on next types:
1. CRUD functions - for work with records in one or more associated tables with pre-validate and transactions.
2. Functions for work with table and associations between them.
3. Migration system(may be union with 2)

### Create functions
All this functions may be add next fields to created record: 
1. created_at - time and timezone
2. (table_name)_count - if exists, save associated records number
3. Something else.

#### create(someObject, [args])
For example, next code
```javascript
someRecord = Record.create({name: "Example", type: "Record"})
```
give next result: 
```
{
  name: "Example", 
  type: "Record",
  created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
}
```
If need create more then single record:
```javascript
someRecords = Record.create([{
      name: "Example", 
      type: "Record"
  },{
     name: "Example2", 
     type: "Record"
}])
```
and result is:
```
[{
  name: "Example", 
  type: "Record",
   created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
},{
  name: "Example2", 
  type: "Record",
   created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
}]
```
#### save(someObject, [args])
Sometimes, object already exist in ram, but haven't record in db. Then use this method. For example:
```javascript
let someObject = Record.new();
someObject.name = "ExampleSave";
someObject.save();
```
Then result is:
 ```
{
  name: "ExampleSave", 
  created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
}
```
If need save array of objects in db, code may be next:
```javascript
let someObjects = [];
for (let i = 0; i < 3; i++) {
   someObjects.push(Record.new({name : "example" + i}));
}
someObjects.save();
```
and get next result:
```
[{
  name: "example0", 
  created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
},{
  name: "example1", 
  created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
},{
  name: "example2", 
  created_at : {
      "time": %timestamp%,
      "zone": %timezone%
  }
  _id: "" //id new record in table 
}]
```
This method work on chainable logic: go to array and call save for each object, then chain transform to single (or more, if object in array is different logic) query to db. Something like this:
```javascript
saveHandler(objects) {

}
```

**Comment by PinkaminaDianePie** [comment]
```javascript
let someObjects = [];
for (let i = 0; i < 3; i++) {
   someObjects.push(Record.new({name : "example" + i}));
}
someObjects.save();
```
Dont get it. We need to add ```save``` method to ```Array```? May be you mean 
```javascript
await Record.save(someObjects); //someObjects are instances of Record in any @@iterable
```
> select()
> Return result with given field

may be change ```select``` name to another, it may be misunderstood as SQL select == find.

> joins() --Probably, not sure

it will be better to move it in separate issue

> Validation

same.

**Comment by stRavens** [comment]
> Dont get it. We need to add save method to Array? May be you mean

Yes, this way better

 > may be change select name to another, it may be misunderstood as SQL select == find.

``` 
SELECT some_fields FROM tablename
```
and there in SELECT section set lists of reading fields. 
By the way, there may add "as" operator from sql syntax: 
```
SELECT somefield AS field FROM sometable 
```
?

**Comment by PinkaminaDianePie** [comment]
> Yes, this way better
Changing builins is bad practice. By the way, what about another @@iterables? You will polute them too? I think static ```save``` metod working with any iterable object will be better.
 
> and there in SELECT section set lists of reading fields. 
In SQL ```select``` mean get items from db, its synonym of ```find()``` in mongo. Operator for specifying fieds must have another name. Why just not to call it ```fields()```?

**Comment by stRavens** [comment]
> Changing builins is bad practice. By the way, what about another @@iterables? You will polute them too? I think static save metod working with any iterable object will be better.
Yes, iterator is good idea. 
 > Why just not to call it fields()
Ok. Also, what about my question about "AS" operator in this function?

**Comment by PinkaminaDianePie** [comment]
> Ok. Also, what about my question about "AS" operator in this function?

it may be related to #7 do we need aliases both in controller and active-record?

**Comment by stRavens** [comment]
> it may be related to #7 do we need aliases both in controller and active-record?
something like
```javascript
@aliases({fieldDb1: 'field1', fieldDb2 : 'field2'})
Record.find()
```
?
What about situation, when fields list and their new names calculate dynamically (rare, but possible)?

**Comment by PinkaminaDianePie** [comment]
> something like
> ```javascript
> @aliases({fieldDb1: 'field1', fieldDb2 : 'field2'})
> Record.find()
> ```

No. es7 decorators can be applied only for class methods or entire class. So we can attach decorator on controller's method definition and change property names after its execution. But we can't attach decorator on method call. I think it can be long discussion so let's do separate issue for that or move discussion to #7.
